activerecord/lib/active_record/migration.rb

```diff
@@ -47,6 +47,26 @@ def initialize
     end
   end
 
+  class NoEnvironmentInSchemaError < MigrationError #:nodoc:
+    def initialize
+      msg = "Environment data not found in the schema. To resolve this issue, run: \n\n\tbin/rake db:migrate"
+      if defined?(Rails.env)
+        super("#{msg} RAILS_ENV=#{::Rails.env}")
+      else
+        super(msg)
+      end
+    end
+  end
+
+  class ProductionRestrictedError < ActiveRecordError#:nodoc:
+    def initialize
+      msg = "You are attempting to run a destructive action against your 'production' database\n"
```

Comment:
Do we want this error message to consider the case where the environment field is blank because they haven't migrated production yet? Is that too much of an edge case?